### ISSUE TITLE
0.13: [Bug]: Setting the version null in pulumi provider does not work

### ISSUE THREAD

**Issue by ishantagarwal-x213235** [garden-io/garden#4832]
## Garden Bonsai (0.13) Bug : Garden is not able identify the local path of pulumi when set the version to null

garden is not able to identify the pulumi when set the version to null in ```project.garden.yaml```
Below is the configuration in project.garden.yaml file,

```providers:
  - name: pulumi
    environments: ["dev"]
    # The version of pulumi to use. Set to `null` to use whichever version of `pulumi` is on your PATH.
    version: null
```

Reported Error
```An error occurred while trying to run 'pulumi/pulumi' (spawn pulumi/pulumi ENOENT). Please make sure 'pulumi/pulumi' is installed and in the $PATH.```

**With version 0.12, it detected the correct path of pulumi**
```
:information_source: x-service [debug] → Spawning '/home/workstation/.pulumi/bin/pulumi stack select telushealth/dev --create' in /home/workstation/thx/workspace/services/21062023/thxcs-shared-services/services/x/app/deployment/pulumi
```

**While with 0.13.7 version, it is not detecting the correct path of pulumi**
```
:information_source: deploy.x-service [debug] → Spawning 'pulumi/pulumi stack select telushealth/dev --create' in /home/workstation/thx/workspace/services/02062023/thxcs-shared-services/services/x/app/deployment/pulumi
```

**Comment by vvagaytsev** [comment]
@ishantagarwal-x213235 the fix has been released in [0.13.9](https://github.com/garden-io/garden/releases/tag/0.13.9), please try it out.

**Comment by ishantagarwal-x213235** [comment]
Thanks for your effort @vvagaytsev , It's working as expected!
